examples/pip_parse_vendored/requirements.bzl

```diff
@@ -7,47 +7,149 @@ from //:requirements.txt
+    ("pip_olefile", "olefile==0.46     --hash=sha256:133b031eaf8fd2c9399b78b8bc5b8fcbe4c31e85295749bb17a87cba8f3c3964"),
+    ("pip_pycparser", "pycparser==2.21     --hash=sha256:8ee45429555515e1f6b185e78100aea234072576aa43ab53aefcae078162fca9     --hash=sha256:e644fdec12f7872f86c58ff790da456218b10f863970249516d60a5eaca77206"),
+    ("pip_pyparsing", "pyparsing==2.4.7     --hash=sha256:c203ec8783bf771a155b207279b9bccb8dea02d8f0c9e5f8ead507bc3246ecc1     --hash=sha256:ef9d7589ef3c200abe66653d3f1ab1033c3c419ae9b9bdb1240a85b024efc88b"),
+    ("pip_requests", "requests==2.28.1     --hash=sha256:7c5599b102feddaa661c826c56ab4fee28bfd17f5abca1ebbe3e7f19d7c97983     --hash=sha256:8fefa2a1a1365bf5520aac41836fbee479da67864514bdb821f31ce07ce65349"),
+    ("pip_urllib3", "urllib3==1.26.13     --hash=sha256:47cc05d99aaa09c9e72ed5809b60e7ba354e64b59c9c173ac3018642d8bb41fc     --hash=sha256:c083dd0dce68dbfbe1129d5271cb90f9447dea7d52097c6e0126120c521ddea8"),
+]
 
+_cluster_mappings = {
+    "oletools": "vbap",
+    "pcodedmp": "vbap",
+}
+
+requirement_clusters = {
+    "vbap": [("oletools", "oletools==0.60.1     --hash=sha256:67a796da4c4b8e2feb9a6b2495bef8798a3323a75512de4e5669d9dc9d1fae31     --hash=sha256:edef92374e688989a39269eb9a11142fb20a023629c23538c849c14d1d1144ea"), ("pcodedmp", "pcodedmp==1.2.6     --hash=sha256:025f8c809a126f45a082ffa820893e6a8d990d9d7ddb68694b5a9f0a6dbcd955     --hash=sha256:4441f7c0ab4cbda27bd4668db3b14f36261d86e5059ce06c0828602cbe1c4278")],
+}
+
+_config = {
+    "download_only": False,
+    "enable_implicit_namespace_pkgs": False,
+    "environment": {},
+    "extra_pip_args": [],
+    "isolated": True,
+    "pip_data_exclude": [],
+    "python_interpreter": "python3",
+    "python_interpreter_target": interpreter,
+    "quiet": True,
+    "repo": "pip",
+    "repo_prefix": "pip_",
+    "timeout": 600,
+}
+
+_annotations = {
+}
+
+# buildifier: disable=function-docstring
 def _clean_name(name):
     return name.replace("-", "_").replace(".", "_").lower()
 
-def requirement(name):
-    return "@pip_" + _clean_name(name) + "//:pkg"
+# buildifier: disable=function-docstring
+def requirement(name, use_clusters = True):
+    cname = _clean_name(name)
+    if cname in _cluster_mappings and use_clusters:
+        return "@pip//:_cluster_{}_dep_{}".format(_cluster_mappings[cname], cname)
+    else:
+        return "@pip_" + cname + "//:pkg"
 
-def whl_requirement(name):
-    return "@pip_" + _clean_name(name) + "//:whl"
+# buildifier: disable=function-docstring
+def whl_requirement(name, use_clusters = True):
+    cname = _clean_name(name)
+    if cname in _cluster_mappings and use_clusters:
+        return "@pip//:whl_" + _cluster_mappings[cname]
+    return "@pip_" + cname + "//:whl"
 
+# buildifier: disable=function-docstring
```

Comment:
I don't think this is new code, but how was it failing buildifier?

Reply:
`buildifier -lint -r .` was barking in `examples/pip_parse_vendored` (6.1.0 from upstream) without this last night while I was trying to avoid iterating on this using CI. I was surprised to be getting function docstring warnings without this because as you note this example originally didn't need these disables. Can run doing without it by CI and see what it says.

Reply:
https://buildkite.com/bazel/rules-python-python/builds/4908#01883047-b108-4b99-b24d-d92a1a8d241b/118-119 recurred in CI when I removed these lines. Putting 'em back.